service/controller/resource/monitoring/verticalpodautoscaler/resource.go

```diff
@@ -204,7 +195,11 @@ func (r *Resource) getMaxMemory(nodes *v1.NodeList) (*resource.Quantity, error)
 		return nil, microerror.Mask(nodeMemoryNotFoundError)
 	}
 
-	q, err := quantityMultiply(nodeMemory, 0.9)
+	// set max `requests` RAM to 80% node RAM.
+	// When setting default limit, make sure max VPA limit won't go higher than available RAM!
+	// because limit grows proportionnaly to requests, and here we compute max requests
+	// So check that PrometheusMemoryLimitCoefficient*MaxMemory < node memory
+	q, err := quantityMultiply(nodeMemory, 0.8)
```

Comment:
Are we sure 0.8 is correct everywhere?

Reply:
I read the end :)

Reply:
Makes the max `requests` to 80% of total node capacity,
but the `limit` will be 96%.
Do you think there's places where we would like to allow it to use more than 96% RAM?
Do you think we should **reserve** more than 80% for prometheus?

This setting seems reasonable to me, and I even think it's a rather small step: we could even be more extreme.
But if I did not convince you, let's talk about it! :grinning:

Reply:
I was also thinking about being a bit more drastic and reserving less on bigger nodes tbh because 96% will prevent other pods from running

Reply:
We can merge this, monitor and reduce if necessary